src/authProvider/eosioAuthProvider.ts

```diff
@@ -0,0 +1,57 @@
+import { createLink, authenticate } from '3id-blockchain-utils'
+import type { LinkProof } from '3id-blockchain-utils'
+import { AccountID } from 'caip'
+
+import AbstractAuthProvider from './abstractAuthProvider'
+import { assert } from '../errors'
+import type { EOSIOProvider } from '../types'
+
+/**
+ *  AuthProvider which can be used for eosio providers that meet the universal authentication interface(UAL)
+ */
+class EOSIOAuthProvider extends AbstractAuthProvider {
+  accountId: string | null
+  provider: EOSIOProvider
+
+  constructor(eosioProvider: EOSIOProvider) {
+    assert.isDefined(eosioProvider, 'Missing EOSIO provider')
+    super('eosio')
+    this.provider = eosioProvider
+    this.accountId = null
+  }
+
+  async init(): Promise<void> {
+    this.accountId = await this._toAccoundId()
+  }
+
+  _toCAIPChainId(chainId: string): string {
+    return `eosio:${chainId.substr(0, 32)}`
+  }
+
+  async _toAccoundId(address?: string): Promise<string> {
```

Comment:
should be typed to return AccountID instead of string? maybe cause of typescript errors below?